src/durablehttprequest.ts

```diff
@@ -0,0 +1,31 @@
+import { TokenSource } from "./tokensource";
+
+/**
+ * Data structure representing a durable HTTP request.
+ */
+export class DurableHttpRequest {
+
+    /**
+     * Creates a new instance of DurableHttpRequest with the
+     * specified parameters.
+     *
+     * @param method The HTTP request method.
+     * @param uri The HTTP request URL.
+     * @param content The HTTP request content.
+     * @param tokenSource The type of OAuth token to add to the request.
```

Comment:
NIT: Not sure I like the language "type of" to refer to token source, but I don't have a great alternative.

Reply:
Yeah, I'm not really sure how to describe this parameter succinctly. Since neither of us have good ideas, I'll just replace "type" with "source" since I can see how "type of" could be misleading.